client/src/components/growth-book/growth-book-wrapper.tsx

```diff
@@ -98,7 +98,11 @@ const GrowthBookWrapper = ({
         clientLocal: clientLocale
       };
 
-      if (user) {
+      // Needs more checks as the user object is not empty but user is not signed in
+      const isSignedIn =
+        user && user.completedChallenges && user.email && user.joinDate;
+
+      if (isSignedIn) {
         userAttributes = {
           ...userAttributes,
           staff: user.email.includes('@freecodecamp'),
```

Comment:
This was throwing since the user object was not empty but it did not have an email